atc/api/teamserver/set.go

```diff
@@ -62,6 +69,16 @@ func (s *Server) SetTeam(w http.ResponseWriter, r *http.Request) {
 	} else if acc.IsAdmin() {
 		hLog.Debug("creating team")
 
+		var warnings []atc.ConfigWarning
+		err = atc.ValidateIdentifier(atcTeam.Name, "team")
+		if err != nil {
+			warnings = append(warnings, atc.ConfigWarning{
```

Comment:
Here, we should always error - we've already determined that the team is new, so we should prevent making a new team with an invalid name